### ISSUE TITLE
request to remove gitea's cpu limit or bump it from 1 --> 3 cpu cores

### ISSUE THREAD

**Issue by neoakris** [defenseunicorns/zarf#454]
**Is your feature request related to a problem? Please describe.**
While working on a bigbang airgap quickstart

I'd run the following command to quickly populate gitea with a copy of repos sourced from repo1
```shell
time ls -1 ~/repos | grep -v bigbang | xargs --replace={item} -P 50 /bin/bash -c 'cd ~/repos/{item}; git remote add gitea https://$GIT_USER:$GIT_PASS@git.bigbang.dev/platform-one/{item}.git; git checkout main; git push gitea main; git push gitea --tags'
```
^-- The above uploads 32 repos sourced from bigbang's airgap artifact to gitea in parallel.

I had to create an nginx ingress controller ingress object with the following annotations to make gitea uploads run smoothly.
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: utility-ingress
  namespace: zarf
  annotations:
    nginx.ingress.kubernetes.io/proxy-body-size: "0" #removes size limit (for HTTPS uploads to gitea using git cli, otherwise there will be a HTTP 413 request entity too large error)
    nginx.ingress.kubernetes.io/proxy-buffering: "off" # (change logic from async to --> sync, and implicitly disables rate limiting)
    nginx.ingress.kubernetes.io/proxy-request-buffering: "off" # (same as above, and implicitly disables rate limiting)
    nginx.ingress.kubernetes.io/proxy-read-timeout: "900" #parallel git repo upload taxes gitea server 
    nginx.ingress.kubernetes.io/proxy-send-timeout: "900" #which slows down responses 
    nginx.ingress.kubernetes.io/proxy-connect-timeout: "900" #increasing timeouts to 15mins, avoids an ephemeral 504 error
...
```

The above annotations allowed me to stop all HTTP error codes (was having HTTP 413 & HTTP 504) error codes before finding the right combination of annotations to configure nginx-ingress controller that ships by default on an RKE2 install. 

The above parallel upload command takes 7 min to upload, where there's no network bottleneck.
k top node shows gitea is maxed out to 1 CPU core, it's limit. 

`k top pod -n=zarf`
```
NAME                                    CPU(cores)   MEMORY(bytes)
zarf-docker-registry-656c4f8c7d-lvlwb   1m           77Mi
zarf-gitea-0                            1000m        964Mi
```
^-- If I run k top pod -n=zarf while the upload is occurring I can see it's CPU bottlenecked. I had to increase the timeouts in the ingress rule to 15min/900seconds, because the cpu bottleneck caused gitea to go over the default timeout (which I think is 2 mins). The whole operation takes ~7 min to upload 32 repos. 

**Describe the solution you'd like**
If you bump the limit up to 3 cpu cores the parallel upload should be able to run faster, atm it's CPU bottlenecked.
(feel free to say no for stability reasons on small hardware deployments, some logic I wrote does work fine with the 1 cpu limit, so there's no real need to change it. That being said I'm going to modify gitea's default cpu limit on my side to speed up parallel upload times. I wanted to point out this use case, and point out why it might be worth bumping the limit, as I suspect someone might not have thought of a need/reason for the limit to be higher than 1 cpu core.)

**Additional context**
I did an imperative edit/patch of cpu limit 1->3, then did an imperative soft reset of gitea and reran the parallel upload with the new cpu setting. 

originally it took 7mins and maxed out the 1 cpu core. after bumping to 3 cpu core. rerunning the parallel upload 
and running watch kubectl top pod -n=zarf (I saw gitea used the full 3 cpu core, and the upload speed went from 7min --> 2min12 seconds.

**Comment by jeff-mccoy** [comment]
7 minutes is a lot longer than my usually repo push was this for Big Bang Core or the whole collection of packages in https://github.com/defenseunicorns/zarf-package-software-factory?  I think it's totally worth looking at the ability to make some of these configurations more flexible, and not just gitea in particular.  Hard-coding for this one use-case wouldn't be ideal because the package system is mostly agnostic to those details.  I definitely don't want to support requiring an ingress again for any of the core zarf capabilities, but we could discuss parallelization for some environments if it makes sense as a separate issue.  However, I'm not actually seeing the same speed concerns you mentioned, could you provide details on what k8s distro you're using and what the hardware specs or vm specs are?

**Comment by neoakris** [comment]
I pushed 32 repos found in the BigBang release artifact called repositories.tar.gz
found on this page. 
https://repo1.dso.mil/platform-one/big-bang/bigbang/-/releases

(My 7 mins are because the logic I used probably isn't as smart as zarf's logic, my logic uploaded all tags of the 32 repos to gitea in parallel. I used a 8 cpu core 32 gb ram VM running on a beefy 32cpu core / 500gb ram bare metal server for testing, the k8s distro was RKE2 with it's default nginx ingress controller)

There's no hard need to implement this, I was able to easily override with a post deployment imperative kubectl patch command. / you could always consider pre-deployment customization could be considered as out of scope / avoid feature creep.

**Comment by jeff-mccoy** [comment]
Copy I'd be interested to see how deploying the Big Bang core package worked for you performance-wise.
